### ISSUE TITLE
Could not start Service[sssd]:

### ISSUE THREAD

**Issue by curly8** [sgnl05/sgnl05-sssd#10]
Testing your module on Centos 6. System does not get in convergent state in a single run. It seems that not all the changes have been applied on a sssd.conf during the first Puppet run. Services array is geting default values and not the values defined in a config hash.

(/Stage[main]/Sssd::Config/File[sssd_config_file]/content) -services = nss, pam, autofs
(/Stage[main]/Sssd::Config/File[sssd_config_file]/content) +services = nss,pam

**Comment by edestecd** [comment]
Hmm, I have several CentOS 6 nodes and have not experienced this.  My config always goes in first and then starts the service, due this ordering in init.pp:
```puppet
anchor { 'sssd::begin': } ->
class { '::sssd::install': } ->
class { '::sssd::config': } ~>
class { '::sssd::service': } ->
anchor { 'sssd::end': }
```

**Comment by edestecd** [comment]
Can you provide a sample of the code you are using to include the main class?
Are you using hiera / PE console / Foreman or simple node definitions?

**Comment by curly8** [comment]
I'm using resource like definition. http://pastebin.com/xZq449hZ

**Comment by edestecd** [comment]
Your example looks pretty similar to mine, except this part:
```puppet
'nss' => {
  'filter_users' => [ 'root','ldap','named','avahi','haldaemon','dbus','radiusd','news','nscd','apache','mysql' ]
},
```

Is that correct?  should that block have some other options?

**Comment by edestecd** [comment]
Other than that I'm not sure...  I may be able to spin up a test box later this eve and play around.
What is your puppet version?  Are you using future parser?

**Comment by curly8** [comment]
That's exactly what I need in this block. My only guessing that the problem is with the Ruby and the way it handles a list of hashes here?

**Comment by edestecd** [comment]
These are puppet hashes, so we should be shielded from the underlying ruby....
Except in the template.  But that does not explain the service starting before the config goes in.

So if you run puppet a second time, it puts in your config and starts the sssd Daemon just fine?
Somehow that scenario rings a bell from when I first started using this module. But I thought it was fixed.
What version of the module are you using.  Is it the latest, 0.2.1, from the forge?

**Comment by curly8** [comment]
The thing is that the config file is created but missing some pieces from the hash, thus preventing the service to start. After 2nd run the file is in place and services starts regularly.
PE2015.2 - everything on default., sssd module  version 0.2.1 from the forge.

**Comment by edestecd** [comment]
I don't have any PE in my environment :(
I think we need some acceptance tests that use beaker with a PE box.

**Comment by edestecd** [comment]
closed by pr #11

**Comment by edestecd** [comment]
closed by pr #11
